### DISCUSSION TITLE
How to distinguish functions and structs with the same name

### DISCUSSION THREAD

**Discussion by luxingxiao** [bytedeco/javacpp#639] [Q&A] [answered]
There is a header file containing structs and functions with the same name, which will cause compilation problems when mapping. How to solve it?

**Comment by saudet** [comment, marked as answer]
I'm pretty sure Java can have classes and methods with the same name as well. Do you have a specific example that fails?

**Reply by luxingxiao** [reply]
Header file fragment:

```
struct idl_name {
  idl_symbol_t symbol;
  char *identifier;
};
typedef struct idl_name idl_name_t;
IDL_EXPORT const idl_name_t *idl_name(const void *node);
```

'idl_name' is is the name of the structure and also the name of the function。
The following error will be reported when compiling:
linux-x86_64/jniJIdlc.cpp:657:114: error: expected primary-expression before ‘)’ token
  657 | static void JavaCPP_com_kernelsoft_jdatabus_global_JIdlc_00024idl_1name_deallocate(void *p) { delete (::idl_name*)p; }

**Reply by saudet** [reply]
I guess we'd have to modify JavaCPP to output "class" as part of the namespace of the struct somehow. Something like a `@Name("class idl_name")` annotation on the Java class might actually work though. Could you give that a try?

**Reply by luxingxiao** [reply]
How to do it?
infoMap.put(new Info("idl_name").javaText("@Name(\"class idl_name\")")) ?

**Reply by saudet** [reply]
I don't think it's currently possible with the Parser to add easily an annotation like that to a class... Could you try to do it manually on a couple of classes and see if it helps?

**Reply by luxingxiao** [reply]
It works! Thank you!
